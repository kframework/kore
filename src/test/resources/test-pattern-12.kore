[]

module TEST-PATTERN-12

  axiom{S}
    \subset{S,S}(X:S,X:S) []

endmodule []
