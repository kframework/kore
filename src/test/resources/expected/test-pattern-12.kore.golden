Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "TEST-PATTERN-12"
            , moduleSentences =
                [ MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ ObjectSortVariable (SortVariable (Id Object "S"))
                        ]
                    , sentenceAxiomPattern =
                        ObjectPattern (CeilPattern Ceil
                            { ceilOperandSort =
                                SortVariableSort (SortVariable (Id Object "S"))
                            , ceilResultSort =
                                SortVariableSort (SortVariable (Id Object "S"))
                            , ceilChild =
                                ObjectPattern (VariablePattern Variable
                                    { variableName = Id Object "X"
                                    , variableSort =
                                        SortVariableSort (SortVariable (Id Object "S"))
                                    })
                            })
                    , sentenceAxiomAttributes = Attributes []
                    })
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }